rfcs/text/????-hook-service.md

```diff
@@ -0,0 +1,134 @@
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+TODO:
+
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- There is the possibility for explosion in values used in `type`, or for blocks to use it to tie their block to a specific embedding application, or set of embedding applications. 
+- The core service which other services build on does not allow sync communication – which may be an issue when it comes to rendering UI. 
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+TODO:
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+# Prior art
+
+[prior-art]: #prior-art
+
+## `editableRef`
+
+[editableRef]: #editableref
+
+The first exploration of this space was the introduction of `editableRef` in the HASH app, currently powering the rich text blocks.
+
+This was designed to avoid blocks being coupled to the particular rich text editing solution implemented in HASH, and in that it succeed.
+
+`editableRef` is a function provided by the embedding application which a block can call with a DOM node rendered by the block, which they wish the embedding application to render rich text present on the `text` field on the block entity into.
+
+It was also used by the embedding application (HASH) to know that any particular block had an editable text section within it. The particular solution HASH uses for rich text editing needs (or needed) to know at the time blocks are loaded that a user can edit text within them. This worked because a block would use `editableRef` by reading it from the properties passed to it, which means the `editableRef` key would be in the block's schema.
+
+`editableRef` is an unsuitable candidate for standardisation for a number of reasons:
+- It does not indicate which block entity property on which to store the rich text data
+- It means that the type of the value in the `text` field is unknown to the block, and may not match the value in the block's schema
+- It has a one time use – you cannot use it to make multiple editable islands within a block (think a blockquote with a caption)
+- The requirement for the use of the feature to show up in the block's schema can make certain kinds of blocks impossible (i.e, blocks that may or may not have editable sections depending on their current state).
```

Comment:
I'm not sure needing to specify a property in the schema which may or may not show up depending on state is impossible - it could be an optional property? Probably this is part of a wider discussion about how these special values are declared in the schema, as flagged in the questions further down.

Reply:
I explained offline that this is connection to where properties may be one of multiple types, and the hook service is only dynamically implemented when a property is a certain type. Essentially the use of the service is dynamic. 

@CiaranMn do you think the text here needs changing after our discussion?